magefile.go

```diff
@@ -639,10 +642,28 @@ func (Test) Unit() error {
 	}, "go", "test", "-cover", "-v", "-race", "./internal/...", "./cmd/...", "./pkg/...")
 }
 
-func (Test) Integration() error {
+func (Test) Integration(ctx context.Context) error {
+	workDir, ok := ctx.Value("workDir").(string)
+	if !ok || workDir == "" {
+		workDir = path.Join(cacheDir, "dev-env")
+	}
+	cluster, err := dev.NewCluster(
+		workDir,
+		dev.WithKubeconfigPath(os.Getenv("KUBECONFIG")), dev.WithSchemeBuilder(runtime.SchemeBuilder{operatorsv1alpha1.AddToScheme, aoapisv1alpha1.AddToScheme}),
+	)
+	if err != nil {
+		return fmt.Errorf("creating cluster client: %w", err)
+	}
+
+	if err := postClusterCreationFeatureToggleSetup(ctx, cluster); err != nil {
+		return fmt.Errorf("failed to perform post-cluster creation setup for the feature toggles: %w", err)
+	}
+	if err := deployFeatureToggles(ctx, cluster); err != nil {
+		return fmt.Errorf("failed to deploy feature toggles: %w", err)
+	}
```

Comment:
Why do we need to deploy feature toggles here? Why not keep the assumption that the current cluster in `KUBECONFIG` already has everything setup?

Reply:
What you're saying is correct during local testing when the developer would've run `mage dev:deploy` to setup things which would include an invokation to `deployFeatureToggles` already.

But in operator-e2e tests, the dev:deploy target isn't called. The operator-e2e tests directly install ADO and call `IntegrationCI` target which in turn calls this `Integration()` method.

For that scenario, it should be ensured that before the actual integration tests are run, the environment should be confirmed to be prepped with the pre-requisites required for testing the feature-toggles.

Moreover for local testing too, I just had a personal preference that even if one doesn't run `mage dev:deploy`, it shouldn't be possible to run `Integration()` tests in an incompliant-feature-toggles manner even if done intentionally.